Tidy the inflammation dataset

katrinleinweber opened the pull request:
See #3. I feel between a rock and a hard place here, about which way to go for the [12. > afternoon "Community-standard data formats, tidy data, data packages"](https://tibhannover.github.io/2018-07-09-FAIR-Data-and-Software/#schedule):

1. [R-eco…dplyr](http://www.datacarpentry.org/R-ecology-lesson/03-dplyr.html) or [r-social…tidyr](http://www.datacarpentry.org/r-socialsci/03-dplyr-tidyr/) are mature & highlight the important aspects :-)

1. The [PANGAEA remix](https://github.com/TIBHannover/2018-07-09-FAIR-Data-and-Software/blob/FAIR-dataset-remix/_episodes_rmd/FAIR-remix-PANGAEA.Rmd) is close to our hearts, but touches only a few aspects. Community-standard variable names are already aligned, so only a few lines of `gather()` & `use_data()` could be appended there.

1. The inflammation dataset can easily be tidied on the technical level, but how does one find community-standard variable names? `patient_ID` & `inflammation_score` seem logical, but are they in any ontology?

#### Ideas for this episode

- [x] convert `reshape2::melt` to `tidyr::gather()`
- [x] include raw data as files & result data as `.rda`
- ~~include all 12 files, but then how to label patients consecutively (1-60, 61-120, etc.)?~~

katrinleinweber committed:
Tidy the inflammation dataset

katrinleinweber committed:
Fix typo

katrinleinweber committed:
Update from reshape2 to tidyr

katrinleinweber committed:
Separate chr conversion from gather()

lwjohnst86 commented on _episodes_rmd/06-tidy-data.Rmd:
```diff
@@ -0,0 +1,73 @@
+teaching: 30
+exercises: 10
+questions:
+- "Which different forms can one dataset have?"
```
"What are possible forms that a dataset can have?"

lwjohnst86 commented on _episodes_rmd/06-tidy-data.Rmd:
```diff
@@ -0,0 +1,73 @@
+exercises: 10
+questions:
+- "Which different forms can one dataset have?"
+- "Which advantages and disadvantages do these forms have?"
```
which -> what

lwjohnst86 commented on _episodes_rmd/06-tidy-data.Rmd:
```diff
@@ -0,0 +1,73 @@
+- "Which features make a dataset more or less reusable?"
+- "How can we add datasets to R packages?"
+objectives:
+- "Use `tidyr::gather()` to convert wide data to its long form."
```
This objective doesn't relate to item 4 of questions. Maybe add to it saying "and adding to package"?

katrinleinweber replied:
More objectives will be added. I have a task list in my staging area ;-)

lwjohnst86 commented on _episodes_rmd/06-tidy-data.Rmd:
```diff
@@ -0,0 +1,73 @@
+objectives:
+- "Use `tidyr::gather()` to convert wide data to its long form."
+keypoints:
+- "Excel & co. incentivise the wide data format, which may spread variables across columns."
```
"Spreadsheets incentivise ..."

lwjohnst86 commented on _episodes_rmd/06-tidy-data.Rmd:
````diff
@@ -0,0 +1,73 @@
+
+```{r inflammation-wide}
+dat <- read.csv(file = "data/inflammation-01.csv", header = FALSE)
+colnames(dat); rownames(dat)
````
Has this `;` been introduced? I tend to prefer one line per command for such short lines of code

katrinleinweber replied:
No. I thought it was necessary to get both outputs into the knit-ed version. But it turns out it isn't, so we'll go with your preference :-)

lwjohnst86 commented on _episodes_rmd/06-tidy-data.Rmd:
````diff
@@ -0,0 +1,73 @@
+
+```{r inflammation-label}
+patient_ID <- paste("patient", sep = "_", seq(60))
+dat_labelled <- cbind(dat, patient_ID)
````
since we use dplyr/tidyr, why not also use `bind_cols` instead of `cbind`?

katrinleinweber replied:
I didn't know about that, thanks for the hint! In this case, though: `Error in cbind_all(x) : Argument 2 must have names`. So, I think we can either stick to `base::cbind` or upgrade the example with `names()` or something like that? What do you think?

Please feel free to add it. I tried `names(patient_ID) <- "patient_ID"` and then `dplyr::bind_cols(dat, patient_ID)` but still get the error.

lwjohnst86 commented on _episodes_rmd/06-tidy-data.Rmd:
```diff
@@ -0,0 +1,73 @@
+---
+title: "Tidying & packaging datasets"
+teaching: 30
```
You think it will take 30 min to cover this? Seems a bit much for so little code. Maybe also include `spread`? And using the `gather(..., -variable_name)` minus ability to exclude a column from the gathering.

katrinleinweber replied:
WIP ;-) I didn't adjust the number after copying from a template.

But yes, `spread` and more about `gather` is a good idea!

katrinleinweber replied:
I didn't include more about this after all, because we can re-use so many other resources if people would like to learn about this. For the context of this inflammation dataset, I mostly want to get the column naming and tidying across.

Also, I would include `git commit`ing the steps in between or in the end, so I think 30min is realistic after all.

lwjohnst86 reviewed (commented):
Given the "WIP" tag, I won't comment on the completeness. Seems good so far. Maybe I'm not understanding, but what exactly are you unsure about doing? I think covering wide to long etc conversion is a very powerful tool, so I agree with covering these.

katrinleinweber committed:
Address review comments

- improve wording
- fix typos
- avoid ;

katrinleinweber committed:
Focus on elegant variable to value conversion

sub would require as.numeric() as well.

katrinleinweber committed:
Follow use_this' convention for raw data

katrinleinweber committed:
Abbreviate preview

katrinleinweber committed:
Rephrase approval & mention AllTrials

katrinleinweber committed:
Explain final data packaging step

katrinleinweber committed:
Refer to external resource for dataset docu

katrinleinweber commented:
Thanks for your comments above, @lwjohnst86 :-)

> what exactly are you unsure about doing?

About which of the 3 options could be the most effective learning experience. `3.` is nearly complete, but that doesn't mean we _have_ to use.

katrinleinweber committed:
Trim whitespace

katrinleinweber committed:
Rephrase task for .R file

katrinleinweber committed:
Draft last steps in package docu & build

katrinleinweber committed:
Rename parameter & files to match common terms

http://www.ontobee.org/search?ontology=&keywords=inflammat*&submit=Search+terms

katrinleinweber committed:
Enrich dataset's metadata

Commit:
Don't hard-code patient_ID length

Commit:
Fix dataset docu formatting

Commit:
Explain importance of good variable names

Commit:
Expand episode summary

Commit:
Minor rephrasing & formatting

katrinleinweber committed:
Refer to further gather() & spread() resources

lwjohnst86 commented on _episodes_rmd/06-tidy-data.Rmd:
```diff
@@ -0,0 +1,147 @@
+
+Just as a good function documentation makes your code more accessible and reusable
+for others and your future self, datasets benefit from a documentation as well.
+Please read through [r-pkgs.had.co.nz/data.html#documenting-data](http://r-pkgs.had.co.nz/data.html#documenting-data)
```
They'll do this on their own I assume?

katrinleinweber replied:
Yes, 5min break for us.

lwjohnst86 commented on _episodes_rmd/06-tidy-data.Rmd:
```diff
@@ -0,0 +1,147 @@
+> > ~~~
+> > #' Inflammation In Patients During Study...
+> > #'
+> > #' @source Pre-registration: \url{http://wwww.alltrials.net/study...}.
```
Any reason not to use markdown syntax in the Roxygen documentation? Use can use it by running `usethis::use_roxygen_md()`.

katrinleinweber replied:
Sorry for the delay in responding! GitHub didn't notify me about your comments :-(

lwjohnst86 commented on _episodes_rmd/06-tidy-data.Rmd:
```diff
@@ -0,0 +1,147 @@
+> > #'  \item{patient_ID}{A factor prepresenting the different patients}
+> > #'  \item{day}{Number of days after start of the study}
+> > #'  \item{inflammatory_response}{Measured daily as described in the methods section of ...}
+> > #' }
```
I don't think there is a `\describe` alternative in markdown, so this would stay if you did use markdown

katrinleinweber replied:
Because of this, and the mention of "LaTeX-like syntax" in an earlier episode, I'd keep it like this first, but I added a note below this: 5b9deca.

lwjohnst86 commented on _episodes_rmd/06-tidy-data.Rmd:
```diff
@@ -0,0 +1,147 @@
+>
+> > ## Solutions
+> > 
+> > 1. Run `roxygen2::roxygenise()`.
```
Or `devtools::document()`

katrinleinweber replied:
Because it's not mentioned anywere else in the lesson anymore, I'd refer learners to `??roxygenise` in case of problems. There, they'll find a note about when to use `devtools::document()`.

lwjohnst86 reviewed (commented):
Looking good!

Commit:
Mention usethis::use_roxygen_md()

Commit:
Fix typos

katrinleinweber commented:
I'm merging this as it is now, because we have to freeze the material at some point I'm going to do it now, and leave #6 to a second iteration of this workshop.
